### PR TITLE
modification for automatic tests run

### PR DESCRIPTION
**Pull request by Mdmorshadurrahman**
Modified code formatting via **pycodestyle** for automatic test ( ignoring error E501(^) )

### CODE REVIEW

File: openmxparser/parser.py [hunk 1]
```diff
@@ -69,6 +69,9 @@ def str_to_sites(string):
         'input_atoms', r'<Atoms.SpeciesAndCoordinates([\s\S]+)Atoms.SpeciesAndCoordinates>',
         sub_parser=input_atoms_parser,
         repeats=False),
+    Quantity('scf_XcType', r'scf.XcType                    (\S+)', repeats=False),
+    Quantity('scf_SpinPolarization', r'scf.SpinPolarization          (\S+)', repeats=False)
```
Comment (ondracka) [inline openmxparser/parser.py:73]: You have multiple spaces here. This won't work when user adds another amount of spaces. Please use \s+ for one or more of any whitespace characters

File: openmxparser/parser.py [hunk 2]
```diff
@@ -97,6 +100,18 @@ def run(self, mainfile: str, archive: EntryArchive, logger):
...
+        scf_SpinPolarizationType = mainfile_parser.get('scf_SpinPolarization')
+        print('scf.SpinPolarization Type is: ',scf_SpinPolarizationType)
+        if scf_SpinPolarizationType.lower() == 'on':
+            method.number_of_spin_channels = 2
+        else :
+            method.number_of_spin_channels = 1
+             
+        print ('number of spin channels is: ',method.number_of_spin_channels) 
+
         md_steps = mainfile_parser.get('md_step')
```
Comment (ondracka) [inline openmxparser/parser.py]: You have multiple print lines there. While it is OK to use that for development and debuging, this should not be here for final commit.

File: openmxparser/parser.py [hunk 3]
```diff
@@ -101,16 +101,16 @@ def run(self, mainfile: str, archive: EntryArchive, logger):
...
         scf_SpinPolarizationType = mainfile_parser.get('scf_SpinPolarization')
-        print('scf.SpinPolarization Type is: ',scf_SpinPolarizationType)
+        print('\n\tscf.SpinPolarization Type is: ',scf_SpinPolarizationType)
         if scf_SpinPolarizationType.lower() == 'on':
             method.number_of_spin_channels = 2
         else :
             method.number_of_spin_channels = 1
              
-        print ('number of spin channels is: ',method.number_of_spin_channels) 
+        print ('\tnumber of spin channels is: ',method.number_of_spin_channels,'\n\n')
```
Comment (ondracka) [inline openmxparser/parser.py]: This is unnecessary commit, the prints should be removed instead.

**Comment by ondracka** [comment]
Hi Joseph,

thanks for the pull request. This is a good start but there are still multiple issues:

1. the commits are not doing what they claim they do in the commit message, for example "modification for automatic tests run" has the structure parsing stuff inside.
2. The tests are not actually fixed. See the run results https://github.com/ondracka/nomad-parser-openmx/runs/2489512273 I tried to make the tests run automatically fro new pull requests, so you can see what is failing but I'm not sure if I managed to do that.
3. There are some debug prints forgotten
4. There are no tests. There should be tests for every addition to the metainfo, see commit https://github.com/ondracka/nomad-parser-openmx/commit/cae18750c59fa7b2aefcd1a595099f87f0fc7d1f for example where tests were added together with the code.
5. Other random minor issues, see inline comments.

File: openmxparser/metainfo.py [hunk 4]
```diff
@@ -30,3 +30,5 @@ class ExampleSCC(SCC):
     # We define an additional example quantity. Use the prefix x_<parsername>_ to denote
     # non common quantities.
     x_example_magic_value = Quantity(type=int, description='The magic value from a magic source.')
+
+    # program_basis_set_type = Quantity(type=str, description= 'should be always "Numeric AOs"')
```
Comment (ondracka) [inline openmxparser/metainfo.py:34]: Don't commit dead code.

**Review by ondracka — CHANGES_REQUESTED** [review]
As far as I can see this commit does not fix the tests but rather adds the structure parsing stuff we did during the last zoom call.

**Comment by ondracka** [comment]
I'm still learning the github review process, so if something is unclear please let me know.

File: openmxparser/parser.py (hunk 1 above)
Comment (Mdmorshadurrahman) [reply openmxparser/parser.py:73]: ok corrected.

File: openmxparser/parser.py (hunk 2 above)
Comment (Mdmorshadurrahman) [reply openmxparser/parser.py]: corrected.

File: openmxparser/parser.py (hunk 3 above)
Comment (Mdmorshadurrahman) [reply openmxparser/parser.py]: Removed.

**Comment by ondracka** [comment]
Any updates here?

**Comment by Mdmorshadurrahman** [comment]
> Any updates here?

As I am also not really good in GitHub, I struggled a bit with branch merging but failed so now I will make a new pull request by today otherwise I might get in more complexity.

**Comment by ondracka** [comment]
Most of this pull request is now either committed through the other pull request or obsolete. I'm closing it now.

**Pull request closed without merging**